instrumentation/opentelemetry-instrumentation-mysql/tests/test_mysql_integration.py

```diff
@@ -103,6 +105,22 @@ def test_instrument_connection(self):
         spans_list = self.memory_exporter.get_finished_spans()
         self.assertEqual(len(spans_list), 1)
 
+    @patch("mysql.connector.connect", new=mock_connect)
+    # pylint: disable=unused-argument
+    def test_instrument_connection_no_op_tracer_provider(self):
+        reset_trace_globals()
+        tracer_provider = trace_api.NoOpTracerProvider()
+        trace_api.set_tracer_provider(tracer_provider)
```

Comment:
You don't have to reset the globals. I think you can use `tracer_provider` arg for `.instrument`